src/sync/app.cpp

```diff
@@ -183,5 +190,30 @@ void App::login_with_credentials(const AppCredentials& credentials,
     }, handler);
 }
 
+void App::log_out(std::function<void (Optional<AppError>)> completion_block) const {
+    std::string route = util::format("%1/session", m_auth_route);
+
+    auto handler = [completion_block, this](const Response& response) {
+        if (auto error = check_for_errors(response)) {
+            return completion_block(error);
+        }
+
+        this->current_user()->log_out();
+
+        return completion_block(util::none);
+    };
+
+    m_config.transport_generator()->send_request_to_server({
+        HttpMethod::post,
+        route,
+        m_request_timeout_ms,
+        {
+            { "Content-Type", "application/json;charset=utf-8" },
+            { "Accept", "application/json" }
```

Comment:
Don't you also need to provide auth tokens to log out? Otherwise that would seem insecure?

Reply:
Good catch. I'm surprised the smoke test worked.

Reply:
We can beef up the integration tests once I get CI to cooperate with the stitch docker image.